src/force_latest_compatible_version.jl

```diff
@@ -0,0 +1,75 @@
+function _force_latest_compatible_version(ctx::Types.Context; target_name = nothing)
```

Comment:
The `_force_latest_compatible_version` seems pretty arbitrarily cut up into 3 functions. Can't you just merge them?

Reply:
Also, it seems excessive for a whole new file for this. Also, we haven't really used `_` prefix anywhere else in Pkg.

Reply:
1. So, in the future, I would like to maybe add the ability to only force the latest compatible version of a strict subset of dependencies. So e.g. the user would pass a list of package names as `Pkg.test(; force_latest_compatible_version = ["Foo", "Bar", "Baz"]`, and then we would only enforce this check for those three packages. That's why I split up `_force_latest_compatible_version` into three functions, so that in the future, there will be less refactoring that I need to do in order to add this feature.
2. I'll move this code into the `src/Operations.jl` source code file.
3. The reason that the `_force_latest_compatible_version` function starts with `_` is that the keyword argument to `Pkg.test` is called `force_latest_compatible_version`. So I had to add the `_` to the function name, otherwise we have a name conflict.

Reply:
I have removed the `_` prefix from all other function names. I had to leave the `_` prefix in the `_force_latest_compatible_version` function name because it conflicts with the name of the `force_latest_compatible_version` kwarg.

Reply:
Okay, I have renamed the `_force_latest_compatible_version` function to `check_force_latest_compatible_version`. Now, this PR do not have any functions that begin with the `_` prefix.

Reply:
@KristofferC Can I mark this as resolved?

I've removed the extra file, and I've removed all the `_` prefixes, so the only outstanding issue is the fact that the function is split up into several different methods, which I'd like to keep to make future refactoring and feature addition easier.

Reply:
@KristofferC marking this as resolved.

Reply:
Just FYI, I ended up merging some of the methods of the `check_force_latest_compatible_version` function after all. So now there are only two methods for `force_latest_compatible_version`.